### PR TITLE
Fix allure-httpclient request without body

### PR DESCRIPTION
**Pull request by useriq**
In the case of receiving apache http response without body, exception occurs: IllegalArgumentException("Body may not be null");
But for response with 304 (Not Modified) code response body must be empty according to RFC: https://restfulapi.net/http-status-codes/ ( This status code is similar to 204 (“No Content”) in that the response body must be empty.

Current PR contains fix of that issue. In case of body absence, "No body is present" will be logged in the allure report.

### CODE REVIEW

File: allure-httpclient/src/main/java/io/qameta/allure/httpclient/AllureHttpClientResponse.java [hunk 1]
```diff
@@ -1,10 +1,6 @@
 package io.qameta.allure.httpclient;
 
-import io.qameta.allure.attachment.AttachmentData;
-import io.qameta.allure.attachment.AttachmentProcessor;
-import io.qameta.allure.attachment.AttachmentRenderer;
-import io.qameta.allure.attachment.DefaultAttachmentProcessor;
-import io.qameta.allure.attachment.FreemarkerAttachmentRenderer;
+import io.qameta.allure.attachment.*;
```
Comment (baev) [inline allure-httpclient/src/main/java/io/qameta/allure/httpclient/AllureHttpClientResponse.java]: revert this change (in our project we are not using wildcard imports)

Comment (useriq) [reply allure-httpclient/src/main/java/io/qameta/allure/httpclient/AllureHttpClientResponse.java]: sorry) reverted

**Comment by baev** [comment]
@useriq also in order to have this merged you need to link your commits with your Github account (see the bottom part of CLAassistant message)

**Comment by baev** [comment]
@useriq thanks!

**Comment by SergiiTrybushnyi** [comment]
Hi there!
Experiencing the same issue, but in Request. We have empty POST requests and when I try to make a request, execution fails on line 47 (entity.writeTo(os);) in AllureHttpClientRequest.class.
It cannot define type of body and throws NPE (as I understand). Changed it to emtpy string "" and works fine now. 
Is it a bug? Or works as expected?
